### PR TITLE
[SPARK-18871][SQL][TESTS] New test cases for IN/NOT IN subquery 4th batch

### PR DESCRIPTION
**Pull request by kevinyu98**
## What changes were proposed in this pull request?

This is 4th batch of test case for IN/NOT IN subquery. In this PR, it has these test files:

`in-set-operations.sql`
`in-with-cte.sql`
`not-in-joins.sql`

Here are the queries and results from running on DB2.

[in-set-operations DB2 version](https://github.com/apache/spark/files/772846/in-set-operations.sql.db2.txt)
[Output of in-set-operations](https://github.com/apache/spark/files/772848/in-set-operations.sql.db2.out.txt)
[in-with-cte DB2 version](https://github.com/apache/spark/files/772849/in-with-cte.sql.db2.txt)
[Output of in-with-cte](https://github.com/apache/spark/files/772856/in-with-cte.sql.db2.out.txt)
[not-in-joins DB2 version](https://github.com/apache/spark/files/772851/not-in-joins.sql.db2.txt)
[Output of not-in-joins](https://github.com/apache/spark/files/772852/not-in-joins.sql.db2.out.txt)

## How was this patch tested?

This pr is adding new test cases. We compare the result from spark with the result from another RDBMS(We used DB2 LUW). If the results are the same, we assume the result is correct.

### CODE REVIEW

**Comment by gatorsmile** [comment]
OK to test

File: sql/core/src/test/resources/sql-tests/results/subquery/in-subquery/in-with-cte.sql.out [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+       FROM   t1
+       WHERE  t1b IN
+              (
+                         SELECT     t2b
+                         FROM       t2
+                         RIGHT JOIN t1
+                         ON         t1c = t2c
+                         LEFT JOIN  t3
+                         ON         t2d = t3d ) AND
+              t1a = "val1b")
```
Comment (gatorsmile) [inline sql/core/src/test/resources/sql-tests/results/subquery/in-subquery/in-with-cte.sql.out]: The style looks strange. Could you adjust them?

Comment (kevinyu98) [inline sql/core/src/test/resources/sql-tests/results/subquery/in-subquery/in-with-cte.sql.out]: sure, I have adjust the style and resubmit. thanks.

**Comment by gatorsmile** [comment]
ok to test

File: sql/core/src/test/resources/sql-tests/results/subquery/in-subquery/in-set-operations.sql.out [hunk 2]
```diff
@@ -0,0 +1,595 @@
...
+-- !query 15 output
+1	8	16	2014-05-05
+1	8	16	2014-05-04
+1	16	12	2014-06-04
+1	16	12	2014-07-04
+1	6	8	2014-04-04
+1	10	NULL	2014-08-04
+1	10	NULL	2014-09-04
+1	10	NULL	2015-05-04
+1	10	NULL	2014-05-04
```
Comment (nsyca) [inline sql/core/src/test/resources/sql-tests/results/subquery/in-subquery/in-set-operations.sql.out]: All the results are equivalent with the ones from DB2.

File: sql/core/src/test/resources/sql-tests/results/subquery/in-subquery/in-with-cte.sql.out [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+                    JOIN       cte1 cte2 ON cte1.t1a = cte2.t1a
+                    RIGHT JOIN cte1 cte3 ON cte1.t1b = cte3.t1b
+                    JOIN cte1 cte4 ON cte1.t1c = cte4.t1c) AND
+         t1c IS NOT NULL
+ORDER BY t1c DESC
+-- !query 12 schema
+struct<t1a:string,t1b:smallint,t1c:int,t1d:bigint,t1h:timestamp>
+-- !query 12 output
+val1b	8	16	19	2014-05-04 01:01:00
+val1c	8	16	19	2014-05-04 01:02:00.001
```
Comment (nsyca) [inline sql/core/src/test/resources/sql-tests/results/subquery/in-subquery/in-with-cte.sql.out]: All the results are equivalent with the ones from DB2.

**Comment by hvanhovell** [comment]
@kevinyu98 @nsyca @dilipbiswal could someone confirm that these results match DB2?

I also think that this PR is almost too large.

File: sql/core/src/test/resources/sql-tests/results/subquery/in-subquery/not-in-joins.sql.out [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+                    INNER JOIN t3
+                    ON         t2b = t3b )
+GROUP BY t1b,
+         t1c,
+         t1d
+HAVING   t1b < sum(t1c)
+-- !query 8 schema
+struct<count(DISTINCT t1a):bigint,t1b:smallint,t1c:int,t1d:bigint>
+-- !query 8 output
+1	6	8	10
```
Comment (nsyca) [inline sql/core/src/test/resources/sql-tests/results/subquery/in-subquery/not-in-joins.sql.out]: All the results are equivalent with the ones from DB2.

**Comment by nsyca** [comment]
It's larger than typical test PRs we submitted for the subquery JIRA but since it's the last test PR, we think we wanted to avoid an additional round of administrative work.

**Comment by gatorsmile** [comment]
retest this please

**Comment by gatorsmile** [comment]
LGTM. Merging to master.

**Comment by kevinyu98** [comment]
@gatorsmile thanks a lot.

**Pull request closed without merging**